### PR TITLE
HPCC-27209 VCPKG Client Tools Support

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by richardkchapman — APPROVED** [review]

File: system/logaccess/CMakeLists.txt [hunk 1]
```diff
@@ -14,6 +14,8 @@
 #    limitations under the License.
 ################################################################################
 
-IF((NOT WIN32) AND USE_ELASTICSTACK_CLIENT)
-  HPCC_ADD_SUBDIRECTORY (ElasticStack)
+IF(NOT CLIENTTOOLS_ONLY)
+  IF((NOT WIN32) AND USE_ELASTICSTACK_CLIENT)
+    HPCC_ADD_SUBDIRECTORY (ElasticStack)
+  ENDIF()
 ENDIF()
```
Comment (GordonSmith) [inline system/logaccess/CMakeLists.txt:21]: @rpastrana is it ok to assume that the elasticstack client can be omitted for client tools?

File: CMakeLists.txt [hunk 2]
```diff
@@ -493,6 +495,7 @@ if(TOP_LEVEL_PROJECT)
 
             set(CPACK_NSIS_MODIFY_PATH "ON")
 
+            if (EXISTS "${SIGN_DIRECTORY}/passphrase.txt")
```
Comment (richardkchapman) [inline CMakeLists.txt:498]: Indentation feels off here

**Review by richardkchapman — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (GordonSmith) [reply CMakeLists.txt:498]: Fixed

**Comment by AttilaVamos** [comment]
@GordonSmith I am sure the regression is not related to this PR.

**Review by rpastrana — APPROVED** [review]
@GordonSmith do any of the clienttool components require access to jlog output?
Currently WsLogAccess, and soon WsWorkunits will be referencing logaccess if those are not part of clienttools, it is safe to exclude elasticlient.

**Comment by GordonSmith** [comment]
@rpastrana I don't know offhand, but even if they did, they would access via ESP?

**Comment by rpastrana** [comment]
> @rpastrana I don't know offhand, but even if they did, they would access via ESP?

Accessing the logs via ESP is fine, the problem would come in if any client tool components required accessing the jlog output directly through the logaccess API. It seems fine to exclude elasticlient from client tools. @GordonSmith

**Review by richardkchapman — APPROVED** [review]

**Pull request merged**